### DISCUSSION TITLE
Cannot find know tool script for ...

### DISCUSSION THREAD

**Discussion by PoloNX** [xmake-io/xmake#5014] [Q&A] [answered]
Hi, I'm using one custom toolchain. I set toolset like this :

```
    set_toolset("cc", "aarch64-none-elf-gcc")
    set_toolset("cxx", "aarch64-none-elf-g++")
    set_toolset("ld", "aarch64-none-elf-g++")
    set_toolset("sh", "aarch64-none-elf-g++")
    set_toolset("ar", "aarch64-none-elf-ar")
    set_toolset("strip", "aarch64-none-elf-strip")
    set_toolset("objcopy", "aarch64-none-elf-objcopy")
    set_toolset("ranlib", "aarch64-none-elf-ranlib")
    set_toolset("as", "aarch64-none-elf-as")
```
    
 and when I build the following package : [glfw](https://github.com/zeromake/xrepo/blob/master/packages/x/xfangfang_glfw/xmake.lua)
 
 I'm getting the following error :
 ``error: ...amdir\core\sandbox\modules\import\core\tool\compiler.lua:37: cannot find known tool script for C:\devkitPro\devkitA64\bin\aarch64-none-elf-as``

 But above there is :
 
 ```
 checking for aarch64-none-elf-as ... C:\devkitPro\devkitA64\bin\aarch64-none-elf-as
checking for the assember (as) ... aarch64-none-elf-as
```

Do you have an idea of why ? (``C:\devkitPro\devkitA64\bin\aarch64-none-elf-as`` exists)

**Comment by waruqi** [comment, marked as answer]
try

```lua
set_toolset("as", "aarch64-none-elf-gcc")
```

we do not use `aarch64-none-elf-as` as `as` directly. gcc can compile asm code.
